### ISSUE TITLE
RPA.Database: Call Stored Procedure use of nextset() seems to be problematic

### ISSUE THREAD

**Issue by emanlove** [robocorp/rpaframework#244]
The placement of the nextset() line [[370](https://github.com/robocorp/rpaframework/blob/6ea6fe480f1ade9c7c56d1bafae7c2ad3e2a1065/packages/main/src/RPA/Database.py#L370)] seems to come before one extracts the results. And thus it drops the results, on a single set, out returning an empty result instead. I believe the placement for it should be here,

```python
            if not PY3K:
                name = name.encode("ascii", "ignore")
            cur.callproc(name, params)
            value = []
            for row in cur:
                value.append(row)
            cur.nextset()                   # <----------
            if not sanstran:
                self._dbconnection.commit()
            return value
```
or possibly after the commit (and maybe even further down in the `finally:`). The reference for this placement came from the PyMySQL [code/documentation](https://pymysql.readthedocs.io/en/latest/modules/cursors.html?highlight=nextset#pymysql.cursors.Cursor.callproc). And from[ the DB-API 2.0 Spec it notes](https://www.python.org/dev/peps/pep-0249/#nextset) that not all modules support this so it may be a per module basis.

**Comment by emanlove** [comment]
I should add for a good test database I am using [sakila MySQL Sample Database](https://dev.mysql.com/doc/sakila/en/) and there are several examples stored procedures found in the [sakila doc](https://downloads.mysql.com/docs/sakila-en.pdf). Besides the example output in the docs I also tried using the query command to call stored procedures as another comparison.

```robotframework
    # Perform stored procedure to produce a table of inventory ID numbers for the copies of the film in stock,
    # returning a count that indicates the number of rows in that table.
    ${inventory_ids}    Query    CALL film_in_stock(1,1,@count);
    ${count}            Query    SELECT @count;
    Log Many    ${inventory_ids}  @{count}
    # or
    @{film_in_stock_params}=  Create List    ${1}    ${1}    @count
    #@{film_in_stock_params}=  Create List    1    1    @count
    ${inventory_ids}    Call Stored Procedure    film_in_stock    ${film_in_stock_params}
    ${count}            Query    SELECT @count;
    Log Many    @{inventory_ids}  @{count}
```

**Comment by emanlove** [comment]
Another possibly helpful reference - https://try2explore.com/questions/11619789

**Comment by emanlove** [comment]
Here are the setup instructions for the test case

1. Install MySQL - https://dev.mysql.com/downloads/installer/
 
2. The downloads for the sakila example can be found at https://dev.mysql.com/doc/index-other.html

3. Installation instructions can be found at https://dev.mysql.com/doc/sakila/en/sakila-installation.html
but I used a slightly different method, as the one presented there did not
work for me (kept reporting couldn't find file) To create the sakila database, I ran the following
```bash
mysql -u <username> -p'<password>' -h <server> -P <host> -D <database> < ./sakila-schema.sql
mysql -u <username> -p'<password>' -h <server> -P <host> -D <database> < ./sakila-data.sql
```

4. The test case requires the following python packages
``` bash
pip install robotframework rpaframework pymysql
```
